app/Services/AdventuresLeagueAdapter.php

```diff
@@ -83,33 +86,77 @@ public function getCharacter($filePath): ?Character
+                } elseif ($isCampaignLogEntry || $isCharacterLogEntry) {
+                    $type = Entry::TYPE_GAME;
+                } else {
+                    $type = Entry::TYPE_DOWNTIME;
+                }
                 $entryData = [
                     'user_id' => Auth::id(),
+                    'adventure_id' => null,
+                    'campaign_id' => null,
                     'character_id' => $characterId,
-                    'date_played' => (array_key_exists(3, $data[$i])) ? $data[$i][3] : now(),
-                    'type' => Entry::TYPE_GAME,
+                    'event_id' => null,
+                    'dungeon_master_id' => null,
+                    'dungeon_master' => (array_key_exists(12, $data[$i])) ? (float)$data[$i][12] : "",
+                    'location' => (array_key_exists(11, $data[$i])) ? (float)$data[$i][11] : "",
+                    'levels' => 0,
+                    'date_played' => ($data[$i][3] == "") ? now() : $data[$i][3],
                     'gp' => (array_key_exists(7, $data[$i])) ? (float)$data[$i][7] : 0,
+                    'downtime' => (array_key_exists(8, $data[$i])) ? (float)$data[$i][8] : 0,
+                    'type' => $type,
                 ];
-                array_push($entries, new Entry($entryData));
-            }
-            // Item importing to be refactored
-            /*else {
-                //create the item and attach it to the last entry
-                $latestEntry = end($entries);
-                if (is_null($latestEntry)) {
-                    continue;
+                $entry = Entry::factory()->create($entryData);
+                $entry->save();
+                array_push($entries, $entry);
+            } else {
+                if (count($entries) == 0) {
+                    $e = Entry::factory()->create([
+                        'user_id' => Auth::id(),
+                        'adventure_id' => null,
+                        'campaign_id' => null,
+                        'character_id' => $characterId,
+                        'event_id' => null,
+                        'dungeon_master_id' => null,
+                        'levels' => 0,
+                        'date_played' => now(),
+                        'gp' => 0,
+                        'downtime' => 0,
+                        'type' => Entry::TYPE_GAME,
+                    ]);
+                    $e->save();
+                    array_push($entries, $e);
                 }
+                $lastEntry = $entries[array_key_last($entries)];
                 $itemData = [
-                    'name' => $data[$i][1],
-                    'rarity' => $data[$i][2],
+                    'entry_id' => $lastEntry->id,
+                    'character_id' => $characterId,
+                    'name' => (array_key_exists(1, $data[$i])) ? $data[$i][1] : "",
+                    //'rarity' => (array_key_exists(2, $data[$i])) ? $data[$i][2] : "common", //need enum
```

Comment:
Ah yeah, this just needs to be added as a property on the `Item` model, since it's already set in the db, something like: 

`App\Models\Item.php`
```php
public const RARITY = ["common", "uncommon", "rare", "very_rare", "legendary"]
```

Then we'd just need some extra function that can constrain a string to one of those options

Reply:
I figured something like this, but when I coded it to take "common" as the default (as seen in the above commented code), it threw an error about truncating the field. A bit of research into it suggested it was because of the enum. That's about as far as i got there.